ShopApp/Data/Repositories/ShopAppCartRepository.swift

```diff
@@ -15,8 +15,8 @@ class ShopAppCartRepository: CartRepository {
         self.dao = dao
     }
 
-    func getCartProductList(callback: @escaping RepoCallback<[CartProduct]>) {
-        callback(dao.getCartProductList(), nil)
+    func getCartProductList() -> [CartProduct] {
```

Comment:
I think we should have the same way with callback and pass error from Dao if error happens

Reply:
There is no error when we fetch list of products with corestore, it will be empty or not empty array (you can look at implementation of `CoreDataDAO`). So I think we can use return type of function instead callback.

Reply:
That's because we have synchronous request to CoreData. Please, rewrite it to asynchronous and handle error.